src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java

```diff
@@ -93,19 +95,116 @@ public void testNoEmptyAnchors() throws IOException {
 
         // No anchor should be empty
         for (Link link:a.getLinks()){
-            assertFalse(link.getDescription().isEmpty());
+            assertFalse(link.getAnchor().isEmpty());
         }
 
         // testing an specific anchor
         for (Link link:a.getLinks()){
             if (link.getId().equals("HTMS_Chakri_Naruebet"))
-                assertEquals(link.getDescription(),"HTMS Chakri Naruebet");
+                assertEquals(link.getAnchor(),"HTMS Chakri Naruebet");
         }
 
+    }
+    
+    @Test
+    public void testNoEmptyWikiIds() throws IOException {
+        Article a = new Article();
+        String mediawiki = IOUtils.getFileAsUTF8String("./src/test/resources/en/Cenozoic");
+        parser.parse(a, mediawiki);
 
+        for(Link l: a.getLinks()){
+            assert(!l.getId().equals(""));
+        }
+    }
+    
+    @Test
+    public void testEmptyLinksShouldBeFiltered() throws IOException {
```

Comment:
I changed them on the nullable indexes (``paragraphId``, ``listId``, etc), but not on ``start`` and ``end`` attributes because they are never supposed to be null.